pkg/sentry/fs/g3doc/fuse.md

```diff
@@ -141,6 +141,57 @@ ops can be implemented in parallel.
 -   a channel that the kernel task will be blocked on when the fd is not
     available.
 
+#### Basic I/O Implementation
+
+Currently we have implemented basic functionalities of read and write
+for our FUSE. We describe the design and ways to improve it here:
+
+##### Basic FUSE Read
+
+The vfs2 expects implementations of `vfs.FileDescriptionImpl.Read()` and
+`vfs.FileDescriptionImpl.PRead()`. When a syscall is made, it will eventually
+reach our implementation of those interface functions located at
+`pkg/sentry/fsimpl/fuse/regular_file.go` for regular files.
+
+After validation checks of the input, sentry sends `FUSE_READ` requests
+to the FUSE daemon. The FUSE daemon returns data after the `fuse_out_header`
+as the responses. Since there is no existing way to directly copy data from
+one userspace memory to another, we create a copy in kernel memory
+of those data. They are represented as a byte slice in the marshalled
+struct. This happens as a common process for all the FUSE responses at this
+moment at `pkg/sentry/fsimpl/fuse/dev.go:writeLocked()`. We then directly
+copy from this intermiediate buffer to the input buffer
+provided by the read syscall.
+
+There is an extra requirement for FUSE: When mouting the FUSE fs,
+the mounter or the FUSE daemon can specify a `max_read` or a `max_pages`
+parameter. They are the upperbound of each `FUSE_READ` request. We
```

Comment:
Elaborate what "upperbound" means. Is this the maximum number of bytes that can be read in a single request?